### PR TITLE
feat: add table width to base trend widget and refactor

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [x] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

Please review in order by commits.

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue [hunk 1]
```diff
@@ -107,7 +107,7 @@ const state = reactive({
             ...refinedFields,
         ];
     }),
-    legends: [] as Legend[],
+    legends: computed<Legend[]>(() => getXYChartLegends(state.data?.results, widgetState.groupBy, props.allReferenceTypeInfo, state.disableReferenceColor)),
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:123]: praise: changed to computed!!!!!! 🎉

File: apps/web/src/services/dashboards/widgets/_components/WidgetDataTable.vue [hunk 2]
```diff
@@ -164,17 +164,18 @@ const getTooltipContents = (item, field:Field):string => {
 };
 
 /* event */
-const handleClickLegend = (index) => {
+const handleClickLegend = (index: number) => {
     // if (props.printMode) return;
-    const _legends = cloneDeep(props.legends);
-    _legends[index].disabled = !_legends[index].disabled;
-    state.proxyLegend = _legends;
+    state.disabledLegends[index] = !state.disabledLegends[index];
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_components/WidgetDataTable.vue:169]: praise: much better...

File: apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue [hunk 3]
```diff
@@ -52,10 +54,15 @@ import { useWidgetLifecycle } from '@/services/dashboards/widgets/_hooks/use-wid
...
+import type { CostAnalyzeResponse, Legend, XYChartData } from '@/services/dashboards/widgets/type';
 
 
-type Data = CostAnalyzeDataModel;
+interface Data {
+    cost_sum: { date: string; value: number }[];
+    _total_cost_sum: number;
+    [groupBy: string]: any;
+}
+type Response = CostAnalyzeResponse<Data>;
```
Comment (skdud4659) [inline apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:65]: ohho,,🧐👍🏻

File: apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue [hunk 4]
```diff
@@ -225,7 +254,7 @@ const handleSelectSelectorType = (selected: SelectorType) => {
     selectedSelectorType.value = selected;
     refreshWidget();
 };
-const handleToggleLegend = (index) => {
+const handleToggleLegend = (index: number) => {
```
Comment (skdud4659) [inline apps/web/src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:257]: 👍🏻👍🏻👍🏻👍🏻👍🏻

**Review by skdud4659 — APPROVED** [review]
LGTM!

**Review by yuda110 — APPROVED** [review]
LGTM. I like your refactoring

File: packages/core-lib/src/index.ts [hunk 5]
```diff
@@ -120,3 +120,35 @@ export const getRGBFromHex = (hex: string): { r: number, g: number, b: number }
...
+            if (Array.isArray(targetArray)) {
+                results.push({
+                    ...d,
+                    [targetArrayFieldKey]: sortBy(targetArray, sortKey),
+                });
+            }
+        });
+    });
+    return results;
+};
```
Comment (sulmoJ) [inline packages/core-lib/src/index.ts:154]: praise: 🙄

**Review by sulmoJ — APPROVED** [review]
LGTM
I just checked code conventions.

**Pull request merged**